swing/src/net/sf/openrocket/gui/main/SimulationPanel.java

```diff
@@ -598,6 +624,216 @@ public void updateEnabledState() {
+			} catch (FileNotFoundException e) {
+				String msg = e.getMessage();
+				JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+			} catch (IOException e) {
+				String msg = e.getMessage();
+				JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+			} finally {
+				if (bufferedWriter != null) {
+					try {
+						bufferedWriter.close();
+					} catch (IOException e) {
+						String msg = e.getMessage();
+						JOptionPane.showMessageDialog(simulationTable.getParent(), msg);
+					}
+				}
+			}
+		}
+
+		private JFileChooser setUpFileChooser() {
+			JFileChooser fch = new JFileChooser();
+			String saveDialogTitle = trans.get("simpanel.pop.export_to_csv.save.dialog.title");
+			String saveButtonText = trans.get("simpanel.pop.export_to_csv.save.button.text");
+			fch.setApproveButtonText(saveButtonText);
+			fch.setDialogTitle(saveDialogTitle);
+			fch.setApproveButtonToolTipText(saveDialogTitle);
+			// set up to filter for .csv's
+			fch.setFileFilter(new FileFilter() {
+				@Override
+				public boolean accept(File f) {
+					if (f.isDirectory()) {
+						return true;
+					} else {
+						return f.getName().toLowerCase().endsWith(".csv");
+					}
+				}
+
+				@Override
+				public String getDescription() {
+					return trans.get("FileHelper.CSV_FILTER");
+				}
+				
+			});
+
+			// default output csv to same name as "the rocket".
+			String documentFileName = document.getRocket().getName();
+			documentFileName += ".csv";
+			fch.setSelectedFile(new File(documentFileName));
+			
+			File currentDir = null;
+			if (!lastSelectedLocation.equals("")) {
+				currentDir = new File(lastSelectedLocation);
+			}
+
+			fch.setCurrentDirectory(currentDir);
+			return fch;
+		}
+
+		@Override
+		public void actionPerformed(ActionEvent arg0) {
+			// one down side is that to change this means you have to export and save at least ONCE!
+			// there is no entry in the preferences to set this a-priori and it is default to a comma
+			String fieldSep = Application.getPreferences().getString(Preferences.EXPORT_FIELD_SEPARATOR, ",");
+
+			if (fieldSep.equals(SPACE)) {
+				fieldSep = " ";
+			} else if (fieldSep.equals(TAB)) {
+				fieldSep = "\t";
+			}
+			Container tableParent = simulationTable.getParent();
+			int modelColumnCount = simulationTableModel.getColumnCount();
+			int modelRowCount = simulationTableModel.getRowCount();
+			populateColumnNameToUnitsHashTable();
+
+			// I'm pretty sure with the enablement/disablement of the menu item under the File dropdown,
+			// that this would no longer be needed because if there is no sim table yet, the context menu
+			// won't show up.   But I'm going to leave this in just in case....
+			if (modelRowCount <= 0) {
+				String msg = trans.get("simpanel.dlg.no.simulation.table.rows");
+				JOptionPane.showMessageDialog(tableParent, msg);
+				return;
+			}
+
+			JFileChooser fch = this.setUpFileChooser();
+			int selectionStatus = fch.showOpenDialog(tableParent);
```

Comment:
Should be `showSaveDialog`. (fixes the issue for macOS not having a naming field)